src/arrayMethodReduce.js

```diff
@@ -1,12 +1,50 @@
 'use strict';
 
-/**
- * Implement method Reduce
- */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc = initialValue !== undefined
+      ? initialValue
+      : arguments.length > 1 ? initialValue
+        : this[0];
+
+    if (acc === this[0]) {
+      for (let i = 1; i < this.length; i++) {
+        acc = callback(acc, this[i], i, this);
+      }
+    } else {
+      for (let i = 0; i < this.length; i++) {
+        acc = callback(acc, this[i], i, this);
+      }
+    }
+
+    return acc;
   };
 }
 
 module.exports = applyCustomReduce;
+
+// function applyCustomReduce() {
```

Comment:
Remove comments. Try it write obvious code that explain all we need without comments)